### ISSUE TITLE
Desktop connection to M1 MacBook not working

### ISSUE THREAD

**Issue by xDeepRedx** [Ylianst/MeshCentral#5067]
**Description**
I've installed the MacOS Agent and restarted the device. The device does not show up on the web interface of the server until I run the agent setup again. Now if I want to connect to the desktop of the device I get the message from MacOS on the device that the agent needs permission to record the screen. I've set the permissions and for screen recording and the whole drive. Now if I want to connect I only see the desktop and the menu bar. But I don't see the bar with the applications and can't remote controll the device. Also tried uninstalling and reinstalling the agent. On a different device with MacOS 11.7.5 Big Sur everything works just fine.

**To Reproduce**
1. Install MacOS Agent on device
2. Restart device
3. You wont see the device on the admin web page
4. Run the MacOS setup again
5. Now you see the device on the web page
6. Connect to the desktop of the device
7. On the device you see the window which asks for permissions for screen recording
8. Grant the permissions
9. Try to connect again
10. You only see the desktop and the menu bar but can't remote control the device

**Expected behavior**
I install the agent on my MacOS device and restart it. After the device booted up and I get to the log in screen, the device should be visible on the web page and I should be able to connect to the desktop of the device and remote control it.

**Screenshots**
Permissions screen which says that you need to grant permissions for audio and screen recording:

![Permission for screen recording](https://user-images.githubusercontent.com/123072975/228803099-c8704aa1-029b-4e05-9ea1-c84d30501ac4.jpeg)

Screen you get when you connect to the desktop of the MacOS device:

![Remote screen](https://user-images.githubusercontent.com/123072975/228803313-d8102f78-1724-45f8-8f67-8bf38e55226f.png)

**Server Software (please complete the following information):**
 - OS: Rocky Linux 9.1
 - Virtualization: None
 - Network: WAN, reverse proxy
 - Version: 1.1.4
 - Node: 16.18.1

**Client Device (please complete the following information):**
 - Device: MacBook Air 2020 M1 processor
 - OS: MacOS Ventura 13.3
 - Network: WAN
 - Browser: Firefox 111.0.1 and Safari 16.4

**Remote Device (please complete the following information):**

**I don't use a remote device.**

 - Device: [e.g. Laptop]
 - OS: [e.g. Windows 10 21H2]
 - Network: [e.g. Local to Meshcentral, Remote over WAN]
 - Current Core Version (if known): [**HINT**: Go to a device then `console` Tab then type `info`]

**Additional context**
Some Information in the config.jason has been change do to anonymizing my domain and other private Information.
The config.jason file, the server and services in general are working perfectly fine with different windows devices and a MacOS
Big Sur 11.7.5 iMac.

**Your config.json file**
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "MongoDb": "mongodb://127.00.1:27017/meshcentral",
    "cert": "subdomain.domain.net",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 443,
    "aliasPort": 443,
    "redirPort": 80,
    "redirAliasPort": 80
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "agentFileInfo": {
              "icon": "wireshark_icon.ico",
              "filedescription": "XXX",
              "_fileversion": "2.0.0.1",
              "internalname": "XXX",
              "legalcopyright": "XXX",
              "originalfilename": "XXX",
              "productname": "MC_Agent",
              "productversion": "v2.0.0.1"
      },
      "_agentCustomization": {
              "_displayname": "Firma",
              "_description": "Test",
              "_companyName": "Firma",
              "_serviceName": "Firmenagent",
              "_fileName": "companyagent"
      },
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "production": false
  }
}

```

**Comment by rrmoore64** [comment]
Same exact issue - Ventura endpoints.  Linux server.  Symptoms and observations match exactly.

**Comment by xDeepRedx** [comment]
> When downloading the agent, did you use the "Apple macOS" agent or the "Linux/BSD/macOS installation program"? It works if you choose the later for "Apple OSX Universal". If you customized your agent, you need to change the file name back to "meshagent". Make it executable and run the installer.
> 
> ```
> chmod +x meshagent
> sudo ./meshagent -install
> ```
> 
> Now your Mac should show up in your meshcentral server.

Thank you for your comment. This works for the permission problem, thank you. But if I restart my MacBook, the device does not show up as online on the MeshCentral server. So the agent does not start automatically after a reboot. Have you experienced the same issue?

**Comment by rosch100** [comment]
Check out [https://github.com/Ylianst/MeshAgent/issues/161](https://github.com/Ylianst/MeshAgent/issues/161) for a solution to the autostart issue.

**Comment by xDeepRedx** [comment]
Removing the .plist file from /Library/LaunchAgents/ worked for me. Now the MacBook shows up when I restart it. Thank you for your help!

**Comment by dinger1986** [comment]
@si458 can close (maybe docs)

**Comment by si458** [comment]
hey gang, just looking at all the mac issues now as somebody kindly donated an M1 to fix bugs!
has anybody found a way to actually get the remote control working on M1 12.7.2 ?
ive tried the mpkg and the binarys and they just dont wanna work?
if i can find a way to get it working i can look into why it needs to be done that way

**Comment by si458** [comment]
update. ive found a work around fix for others if intrested in trying
install the binary as universal, then BEFORE you connect to the device the first time, run `agentupdate` from the console tab
this will force the meshagent to redownload the arm version and use that instead of the universal one
then when you connect, it will ask for screen record permission, so give it and it works!
you sadly still have to disconnect and then reconnect for it to ask for control permission too,
but working on a fix for that!

**Comment by theoroborus** [comment]
> update. ive found a work around fix for others if intrested in trying install the binary as universal, then BEFORE you connect to the device the first time, run `agentupdate` from the console tab this will force the meshagent to redownload the arm version and use that instead of the universal one then when you connect, it will ask for screen record permission, so give it and it works! you sadly still have to disconnect and then reconnect for it to ask for control permission too, but working on a fix for that!

I tried this, but I don't get the remote access tab despite installing the universal package with remote control and updating it. I did the following on M1 Max :

wget -O meshagent "https://mesh*(long url)*installflags=0&meshinstall=10005"
sudo /Users/boris/Downloads/meshagent -update

In Activity Monitor I see : 
<img width="637" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/91cc343c-0717-4016-9338-9f03a38a7856">

**Comment by si458** [comment]
@theoroborus you seem to have 2 agents installed there?
the `meshagent_osx64` i believe is the old pkg version
also you dont need to run `-update`, you should be able to just run `agentupdate` from the console tab of the device
so long as the device is listed in the web ui first!

you might need to uninstall all agents from the device and start afresh

**Comment by theoroborus** [comment]
I rebooted and deleted the remaining meshagent, rebooted again and made sure none were present in the activity monitor.

I ran : 
_brew install zenity_
then : 
_wget -O meshagent "https://mesh***tallflags=0&meshinstall=10005"_
<img width="423" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/dfbbda05-7bf5-4b5a-ac50-274557083e9b">
Then : 
_sudo /Users/boris/meshagent_ 

Yet I get :  
_The graphical version of this installer cannot run on this system.
Try installing/updating Zenity, and run again.

You can run the text version from the command line with the following command(s): 
./meshagent -connect
./meshagent -install
./meshagent -install --installPath="/alternate/path"_

So I ran : 
 _sudo /Users/boris/Desktop/meshagent -install_

And after reconnecting 3x and enabling meshagent in the settings at the 3 different places, it worked.

**Comment by si458** [comment]
@theoroborus yes the is no GUI for the binary so the is no need for the zenity (but nice to have)
you just use the `meshagent -install` as you did to install it!
yes we are aware of the disconnect/reconnect permission bugs, 
it has been fixed in theory, we just need to build new agents and then a lot of testing!

**Comment by si458** [comment]
@theoroborus yes the bug is basically `launchdaemons` and `launchagents` CANNOT have the same name in newer OSX versions,
so the fix is to simple change the Label of your `meshagent.plist` in `LaunchAgents` to `meshagent-launchagent` and then restart your mac!

**Comment by si458** [comment]
@theoroborus sounds like a plan!
we do have some sort of info in the docs but its very basic/missing like hints so you could add the steps/pictures/etc?
https://ylianst.github.io/MeshCentral/meshcentral/agents/#apple-macos-binary-installer
https://github.com/Ylianst/MeshCentral/blob/master/docs/docs/meshcentral/agents.md

we do have a pictures here for the permissions but again no real steps
https://ylianst.github.io/MeshCentral/meshcentral/faq/#mac-clients

**Comment by theoroborus** [comment]
As promised :  

Install MeshCentral Client on ARM based OSX Sonoma : 

The video :
https://youtu.be/A71gioGHLWo

The steps :

- Download the "Apple OSX Universal" client.
<img width="393" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/0bf28c00-dfd5-4779-aedb-57f397e3e73b">

- Open the terminal of your target machine's and type : 
`chmod +x "path to meshagent"`

- And then sudo your 
`sudo "path to meshagent"`

- You should get a warning pop up saying "meshagent" cannot be openned because the developer can not be identified :
<img width="253" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/d46cb3e1-c30f-42f5-b217-c856b2b07e0b">

- Go in Security Settings and click on "Allow anyway" :
<img width="693" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/d4c047d1-44bd-4c92-8adf-ce4c44d15a88">

- Run again : 
`sudo "path to meshagent"`
<img width="263" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/d53963b0-8def-4d42-b638-d6e622c3b3a1">

- Then : 
`sudo "path to meshagent" -install`
`sudo "path to meshagent" -update`

- Edit the file mesgagent.plist in the folder /Library/LaunchAgents :  
<img width="663" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/375d5abd-0a75-4d88-a3d1-d72a19ec8b4a">

- Try to access the desktop of your target from meshcentral. You'll first be facing an empty desktop.
On your target, you'll see a pop up saying "meshagent" would like to record this computer's screen and audio.
<img width="464" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/586b18db-d760-420c-8dd9-abf62e7a66cf">
Open system settings and allow it.
Then disconnect from your target.

- Try to access the desktop of your target from meshcentral a second time.
You'll see a pop up saying "meshagent" would like to control this computer using accessibility features. 
<img width="462" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/5c4e411c-15c8-43cd-b82a-831244eb7799">
Open system settings and allow it as well.

Reboot your target.
MeshAgent should start, and your target should be fully accessible.
You can hide the notifications related to the agent in Notification's settings.

**Comment by theoroborus** [comment]
As promised :  

Install MeshCentral Client on ARM based OSX Sonoma : 

The video :
https://youtu.be/A71gioGHLWo

The steps :

- Download the "Apple OSX Universal" client.
<img width="393" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/0bf28c00-dfd5-4779-aedb-57f397e3e73b">

- Open the terminal of your target machine's and type : 
`chmod +x "path to meshagent"`

- And then sudo your 
`sudo "path to meshagent"`

- You should get a warning pop up saying "meshagent" cannot be openned because the developer can not be identified :
<img width="253" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/d46cb3e1-c30f-42f5-b217-c856b2b07e0b">

- Go in Security Settings and click on "Allow anyway" :
<img width="693" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/d4c047d1-44bd-4c92-8adf-ce4c44d15a88">

- Run again : 
`sudo "path to meshagent"`
<img width="263" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/d53963b0-8def-4d42-b638-d6e622c3b3a1">

- Then : 
`sudo "path to meshagent" -install`
`sudo "path to meshagent" -update`

- Edit the file mesgagent.plist in the folder /Library/LaunchAgents :  
<img width="663" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/375d5abd-0a75-4d88-a3d1-d72a19ec8b4a">

- Try to access the desktop of your target from meshcentral. You'll first be facing an empty desktop.
On your target, you'll see a pop up saying "meshagent" would like to record this computer's screen and audio.
<img width="464" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/586b18db-d760-420c-8dd9-abf62e7a66cf">

Open system settings and allow it.
Then disconnect from your target.

- Try to access the desktop of your target from meshcentral a second time.
You'll see a pop up saying "meshagent" would like to control this computer using accessibility features. 
<img width="462" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/37216326/5c4e411c-15c8-43cd-b82a-831244eb7799">
Open system settings and allow it as well.

Reboot your target.
MeshAgent should start, and your target should be fully accessible.
You can hide the notifications related to the agent in Notification's settings.

**Comment by DarkxPunk** [comment]
https://github.com/Ylianst/MeshCentral/issues/5067#issuecomment-2016527970 your solution seems work, if not quite long winded of a solution. It does however create two meshagent entries within Login Items which is strange and I can't seem to resolve.

**Comment by duke8804** [comment]
Im getting there is nothing to install

<img width="632" alt="Image" src="https://github.com/user-attachments/assets/663af585-e89a-434f-b3fa-4ff8c3c1812b" />
